dbusmock/testcase.py

```diff
@@ -28,6 +30,260 @@
+
+
+class BusType(enum.Enum):
+    SESSION = "session"
+    SYSTEM = "system"
+
+    @staticmethod
+    def from_bool(system_bus: bool) -> "BusType":
+        '''
+        Helper method for legacy methods that use a system_bus=True argument.
+        '''
+        return BusType.SYSTEM if system_bus else BusType.SESSION
+
+    @property
+    def environ(self) -> Tuple[str, Optional[str]]:
+        '''
+        Returns the name and value of this bus' address environment variable
+        '''
+        env = f'DBUS_{self.value.upper()}_BUS_ADDRESS'
+        value = os.environ.get(env)
+        return env, value
+
+    def get_connection(self) -> dbus.bus.Connection:
+        '''
+        Get dbus.bus.BusConnection() object to this bus.
+
+        This uses the current environment variables for this bus (if any) and falls back
+        to dbus.SystemBus() or dbus.SessionBus() otherwise.
+
+        This is preferrable to dbus.SystemBus() and dbus.SessionBus() as those
+        do not get along with multiple changing local test buses.
+        '''
+        _, val = self.environ
+        if val:
+            return dbus.bus.BusConnection(val)
+        elif self == BusType.SYSTEM:
+            return dbus.SystemBus()
+        else:
+            return dbus.SessionBus()
+
+    def reload_configuration(self):
+        '''
+        Notify this bus that it needs to reload the configuration
+        '''
+        bus = self.get_connection()
+        dbus_obj = bus.get_object('org.freedesktop.DBus', '/org/freedesktop/DBus')
+        dbus_if = dbus.Interface(dbus_obj, 'org.freedesktop.DBus')
+        dbus_if.ReloadConfig()
+
+    def wait_for_bus_object(self, dest: str, path: str, timeout: int = 600):
+        '''Wait for an object to appear on D-Bus
+
+        Raise an exception if object does not appear within one minute. You can
+        change the timeout with the "timeout" keyword argument which specifies
+        deciseconds.
+        '''
+        bus = self.get_connection()
+
+        last_exc = None
+        # we check whether the name is owned first, to avoid race conditions
+        # with service activation; once it's owned, wait until we can actually
+        # call methods
+        while timeout > 0:
+            if bus.name_has_owner(dest):
+                try:
+                    p = dbus.Interface(bus.get_object(dest, path),
+                                       dbus_interface=dbus.INTROSPECTABLE_IFACE)
+                    p.Introspect()
+                    break
+                except dbus.exceptions.DBusException as e:
+                    last_exc = e
+                    if '.UnknownInterface' in str(e):
+                        break
+
+            timeout -= 1
+            time.sleep(0.1)
+        if timeout <= 0:
+            assert timeout > 0, f'timed out waiting for D-Bus object {path}: {last_exc}'
+
+
+class MockedDBus:
+    '''A D-Bus daemon instance that prepresents either a session or a system bus.
```

Comment:
"p"represents typo. I don't mind the various typos in the commit messages much, but it's more important in the docstrings.

Reply:
I think I fixed them all now, or at least I've stared at it for long enough that I no longer see them...